internal/oci/oci.go

```diff
@@ -0,0 +1,93 @@
+package wkg
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/regclient/regclient"
+	"github.com/regclient/regclient/types/manifest"
+	"github.com/regclient/regclient/types/ref"
+)
+
+// IsOCIPath checks if a given path is an OCI path
+func IsOCIPath(path string) bool {
+	_, err := ref.New(path)
+	return err == nil
+}
+
+// PullWIT fetches an OCI artifact from a given OCI path
+// It invokes "regclient" APIs to pull the artifact and then
+// processes it with `wasm-tools`.
+// the WIT files are stored in the `<out>` directory
+func PullWIT(ctx context.Context, path, out string) error {
+	r, err := ref.New(path)
+	if err != nil {
+		return fmt.Errorf("failed to parse ref: %v", err)
+	}
+
+	rc := regclient.New()
```

Comment:
I love the idea of integrating directly and understand wanting to avoid another CLI dependency. However, I would recommend using the same config as `wkg` for fetching and loading credentials (which I don't know if `regclient` currently supports, but that should be a requirement as well for those with private registries): https://github.com/bytecodealliance/wasm-pkg-tools/pull/79 (this will be merged soon, just dropping here so you can see the documentation on the config file and how to fetch from the correct OCI registry using the registry.json file

Reply:
Is it possible to integrate `wkg` features directly into wasm-tools? This will benefit this project since it's already shelling out `wasm-tools` for generating components.

Reply:
We'll have to see on that. I'm not opposed, but a lot of people think wasm-tools has gotten far too big/bloated (which I think is a fair assessment). So I'm not completely sure what would happen here

Reply:
It has less functionality than the "go" binary, so not sure I'd call it bloated!

I'd support moving wkg into wasm-tools.

Reply:
Will the config file be supported by standard OCI tools like `regclient` and `oras`? I am asking this question because I think the responsibility of parsing a standardized config file should belong to tools in that field, and should be out of scope for this repo. If the answer is yes, then I think it's fine for merging this PR and we can land new versions of `regclient` later for it.

Reply:
The config file I am mentioning is separate from oci or warg and is specific to wasm registry tooling. So it would belong in here and not in oras or regclient. I can clarify what we're doing here and why and what the plan is for standardizing around "locking" the fetched wit packages next week in the SIG Go meeting.

As for the wasm-tools thing, I think we're going to try and have a convo about that in the component model meeting.

Reply:
I think we will eventually get to implement the wkg config file in Go, but for now, I am okay with using regclient to get the WIT package from an OCI reigistry. Will revisit this later